src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -193,6 +198,38 @@ protected async Task<AuthenticationResult> CacheTokenResponseAndCreateAuthentica
                 msalTokenResponse.SpaAuthCode);
         }
 
+        /// <summary>
+        /// Determines if the token response should be saved in the cached.
+        /// This may be ovveritten by the child types that perform additional checks.
+        /// Token responses recieved from brokers should not be cached for example.
```

Comment:
I don't think we need a virtual method here. We should simply access tokens WAM irrespective of flow. Today WAM can do InteractiveRequest and SilentRequest, and we also plan to add support for ROPC.